### PR TITLE
benchmark memory usage

### PR DESCRIPTION
**Pull request by TomNicholas**
This approach seems to mostly work

```
D select name, projected_memory from test_run;
┌─────────────────────────────────────────────────────┬──────────────────┐
│                        name                         │ projected_memory │
│                       varchar                       │      double      │
├─────────────────────────────────────────────────────┼──────────────────┤
│ test_quad_means[configs/local_single-threaded.yaml] │        1009.6192 │
└─────────────────────────────────────────────────────┴──────────────────┘
D 
```

except that the measured_mem doesn't appear to be being populated - if I print the `events_df` I get 

```
             name  num_tasks  ... peak_measured_mem_start peak_measured_mem_end
0   create-arrays          1  ...                    None                  None
1   create-arrays          1  ...                    None                  None
2   create-arrays          1  ...                    None                  None
3   create-arrays          1  ...                    None                  None
4          op-003          1  ...                    None                  None
5          op-003          1  ...                    None                  None
...
```

### CODE REVIEW

File: tests/utils.py [hunk 1]
```diff
@@ -26,14 +27,17 @@ def run(
         benchmarks,
     ):
 
-    with benchmarks:
+    history = HistoryCallback()
+
+    with benchmarks(history):
```
Comment (TomNicholas) [inline tests/utils.py:32]: Does it make sense to have the instantiated history callback object be the one passed around here? This is replacing the `client` in coiled benchmarks.

**Comment by tomwhite** [comment]
> the measured_mem doesn't appear to be being populated

The default (single-threaded) runner doesn't measure memory. Probably simplest to switch to `threads` to see if it works.

File: tests/utils.py [hunk 2]
```diff
@@ -27,13 +28,17 @@ def run(
         **kwargs
     ):
 
-    with benchmarks:
+    callbacks = kwargs.get('callbacks', [])
+    history = HistoryCallback()
+    callbacks.append(history)
+
+    with benchmarks(history):
```
Comment (TomNicholas) [inline tests/utils.py:35]: To merge the changes from `main` I had to slightly awkwardly append the history callback to any other callbacks already provided.

**Comment by TomNicholas** [comment]
Using the async executor this seems to work

```
D select name, peak_memory from test_run;
┌──────────────────────────────────────────────────────────────────────┬─────────────┐
│                                 name                                 │ peak_memory │
│                               varchar                                │   double    │
├──────────────────────────────────────────────────────────────────────┼─────────────┤
│ test_quad_means[configs/local_single-threaded.yaml]                  │             │
│ test_quad_means[configs/local_single-threaded.yaml]                  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-50]   │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-500]  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-5000] │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-50]   │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-500]  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-5000] │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-50]   │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-500]  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-5000] │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-50]   │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-500]  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-5000] │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-50]   │ 3700.178944 │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-500]  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-50]   │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-500]  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-5000] │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-50]   │  3642.81856 │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-500]  │             │
│ test_quadratic_means_xarray[configs/local_single-threaded.yaml-5000] │             │
├──────────────────────────────────────────────────────────────────────┴─────────────┤
│ 22 rows                                                                  2 columns │
└────────────────────────────────────────────────────────────────────────────────────┘
```

so I added it as another (commented-out) option.

**Pull request merged**